jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -605,6 +608,57 @@ public void writeTo(OutputStream outputStream, long start, long count)
         }
     }
 
+    private void skipTo(SeekableByteChannel channel, ByteBuffer buffer, long skipTo) throws IOException
+    {
+        try
+        {
+            if (channel.position() != skipTo)
+            {
+                channel.position(skipTo);
+            }
+        }
+        catch (UnsupportedOperationException e)
+        {
+            final int NO_PROGRESS_LIMIT = 3;
```

Comment:
This code is kind of duplicated.

Reply:
It's very close, kind of, yeah.
I hope to remove this duplication entirely within Jetty 10. (by implementing Issue #3924)
It's a bit difficult to do in Jetty 9.4.x without breaking backwards compat.